runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexProcessFnOperator.java

```diff
@@ -0,0 +1,191 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.apex.translation.operators;
+
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.common.util.BaseOperator;
+import com.esotericsoftware.kryo.serializers.FieldSerializer.Bind;
+import com.esotericsoftware.kryo.serializers.JavaSerializer;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Iterables;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import org.apache.beam.runners.apex.ApexPipelineOptions;
+import org.apache.beam.runners.apex.translation.utils.ApexStreamTuple;
+import org.apache.beam.runners.core.KeyedWorkItem;
+import org.apache.beam.runners.core.KeyedWorkItems;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.transforms.windowing.WindowFn;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.KV;
+import org.joda.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Apex operator for simple native map operations.
+ */
+public class ApexProcessFnOperator<InputT> extends BaseOperator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ApexProcessFnOperator.class);
+  private boolean traceTuples = false;
+  @Bind(JavaSerializer.class)
+  private final ProcessFn<InputT> fn;
+
+  public ApexProcessFnOperator(ProcessFn<InputT> fn, boolean traceTuples) {
+    super();
+    this.traceTuples = traceTuples;
+    this.fn = fn;
+  }
+
+  @SuppressWarnings("unused")
+  private ApexProcessFnOperator() {
+    // for Kryo
+    fn = null;
+  }
+
+  private final transient OutputEmitter<ApexStreamTuple<? extends WindowedValue<?>>> outputEmitter =
+      new OutputEmitter<ApexStreamTuple<? extends WindowedValue<?>>>() {
+    @Override
+    public void emit(ApexStreamTuple<? extends WindowedValue<?>> tuple) {
+      if (traceTuples) {
+        LOG.debug("\nemitting {}\n", tuple);
+      }
+      outputPort.emit(tuple);
+    }
+  };
+
+  /**
+   * Something that emits results.
+   * @param <T>
+   */
+  public interface OutputEmitter<T> {
+    void emit(T tuple);
+  };
+
+  /**
+   * The processing logic for this operator.
+   *
+   * @param <InputT>
+   */
+  public interface ProcessFn<InputT> extends Serializable {
```

Comment:
I was very confused when reviewing this file, because this clashes with SplittableParDo.ProcessFn. Perhaps rename to something else?